src/System.Net.Requests/src/System/Net/TimerThread.cs

```diff
@@ -287,7 +287,7 @@ private class InfiniteTimerQueue : Queue
         /// </summary>
         private class TimerNode : Timer
         {
-            private TimerState _timerState;
+            private volatile TimerState _timerState;
```

Comment:
Why is this needed?  What race condition is it preventing?

Reply:
Thanks, I see that, but why is the fence supplied by the lock exited immediately after insufficient? This is a primitive value, not an object being published that could be read before fully initialized. Hence my question... what's the actual race condition here being fixed?